src/ripple/core/tests/Config.test.cpp

```diff
@@ -346,10 +448,250 @@ port_wss_admin
+            } catch (std::runtime_error& e) {
+                error = e.what();
+            }
+            expect (error == expectedError);
+        }
+        {
+            // load validators and quorum from config
+            Config c;
+            std::string toLoad(R"rippleConfig(
+[validators]
+n949f75evCHwgyP4fPVgaHqNHxUVN15PsJEZ3B3HnXPcPjcZAoy7
+n9MD5h24qrQqiyBC8aeqqCWvpiBiYQ3jxSr91uiDvmrkyHRdYLUj
+n9L81uNCaPgtUJfaHh89gmdvXKAmSt5Gdsw2g1iPWaPkAHW5Nm4C
+
+[validator_keys]
+nHUhG1PgAG8H8myUENypM35JgfqXAKNQvRVVAFDRzJrny5eZN8d5
+nHBu9PTL9dn2GuZtdW4U2WzBwffyX9qsQCd9CNU4Z5YG3PQfViM8
+
+[validation_quorum]
+4
+)rippleConfig");
+            c.loadFromString (toLoad);
+            expect (c.legacy ("validators_file").empty ());
+            expect (c.section (SECTION_VALIDATORS).values ().size () == 3);
+            expect (c.section (SECTION_VALIDATOR_KEYS).values ().size () == 2);
+            expect (c.VALIDATION_QUORUM == 4);
+        }
+        {
+            // load from specified [validators_file] absolute path
+            int const quorum = 3;
+            detail::ValidatorsTxtGuard vtg (
+                *this, "test_cfg", "validators.cfg", std::to_string(quorum));
+            expect (vtg.validatorsFileExists ());
+            Config c;
+            boost::format cc ("[validators_file]\n%1%\n");
+            c.loadFromString (boost::str (cc % vtg.validatorsFile ()));
+            expect (c.legacy ("validators_file") == vtg.validatorsFile ());
+            expect (c.section (SECTION_VALIDATORS).values ().size () == 5);
+            expect (c.section (SECTION_VALIDATOR_KEYS).values ().size () == 3);
+            expect (c.VALIDATION_QUORUM == quorum);
+        }
+        {
+            // load from specified [validators_file] file name
+            // in config directory
+            int const quorum = 3;
+            std::string const valFileName = "validators.txt";
+            detail::ValidatorsTxtGuard vtg (
+                *this, "test_cfg", valFileName, std::to_string(quorum));
+            detail::RippledCfgGuard rcg (
+                *this, "test_cfg", "", valFileName);
+            expect (vtg.validatorsFileExists ());
+            expect (rcg.configFileExists ());
+            auto& c (rcg.config ());
+            expect (c.legacy ("validators_file") == valFileName);
+            expect (c.section (SECTION_VALIDATORS).values ().size () == 5);
+            expect (c.section (SECTION_VALIDATOR_KEYS).values ().size () == 3);
+            expect (c.VALIDATION_QUORUM == quorum);
+        }
+        {
+            // load from specified [validators_file] symlink
+            // in config directory
+            int const quorum = 3;
+            detail::ValidatorsTxtGuard vtg (
+                *this, "test_cfg", "validators.txt", std::to_string(quorum));
+            path const valSymlink = current_path () / "test_cfg/validators.cfg";
+            create_symlink (vtg.validatorsFile (), valSymlink);
```

Comment:
Previously, the validators file was not allowed to be a symlink. This is testing the change.